sql/jobs: TestRegistryLifecycle failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/26a06a661e7c1ce63f3077269005472182286ca4

Parameters:
```
TAGS=
GOFLAGS=-race
```

Stress build found a failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=596181&tab=buildLog

```
I180411 07:23:05.531022 43 util/stop/stopper.go:470  quiescing; tasks left:
1      [async] transport racer
E180411 07:23:05.581750 258 sql/jobs/registry.go:272  error while adopting jobs: node unavailable; try another peer
I180411 07:23:05.597587 266 kv/transport_race.go:67  transport race promotion: ran 21 iterations on up to 1532 requests
```

--- a/pkg/sql/jobs/jobs_test.go
+++ b/pkg/sql/jobs/jobs_test.go
@@ -171,7 +171,7 @@ func TestRegistryLifecycle(t *testing.T) {
 
 	check := func(t *testing.T) {
 		t.Helper()
-		if err := retry.ForDuration(time.Second*2, func() error {
+		if err := retry.ForDuration(time.Second*5, func() error {
 			lock.Lock()
 			defer lock.Unlock()
 			if e != a {
@@ -183,8 +183,10 @@ func TestRegistryLifecycle(t *testing.T) {
 		}
 	}
 	clear := func() {
+		lock.Lock()
 		a = Counters{}
 		e = Counters{}
+		lock.Unlock()
 	}
 
 	resumeCh := make(chan error)

